contracts/src/L1/rollup/ScrollChain.sol

```diff
@@ -441,56 +568,181 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
      * Internal Functions *
      **********************/
 
+    function _commitChunksV0(
+        uint256 _totalL1MessagesPoppedOverall,
+        bytes[] memory _chunks,
+        bytes calldata _skippedL1MessageBitmap
+    ) internal view returns (bytes32 _dataHash, uint256 _totalL1MessagesPoppedInBatch) {
+        uint256 _chunksLength = _chunks.length;
+
+        // load `dataPtr` and reserve the memory region for chunk data hashes
+        uint256 dataPtr;
+        assembly {
+            dataPtr := mload(0x40)
+            mstore(0x40, add(dataPtr, mul(_chunksLength, 32)))
+        }
+
+        // compute the data hash for each chunk
+        for (uint256 i = 0; i < _chunksLength; i++) {
+            uint256 _totalNumL1MessagesInChunk;
+            (_dataHash, _totalNumL1MessagesInChunk) = _commitChunkV0(
+                _chunks[i],
+                _totalL1MessagesPoppedInBatch,
+                _totalL1MessagesPoppedOverall,
+                _skippedL1MessageBitmap
+            );
+            unchecked {
+                _totalL1MessagesPoppedInBatch += _totalNumL1MessagesInChunk;
+                _totalL1MessagesPoppedOverall += _totalNumL1MessagesInChunk;
+            }
+            assembly {
+                mstore(dataPtr, _dataHash)
+                dataPtr := add(dataPtr, 0x20)
+            }
+        }
+
+        assembly {
+            let dataLen := mul(_chunksLength, 0x20)
+            _dataHash := keccak256(sub(dataPtr, dataLen), dataLen)
+        }
+    }
+
+    function _commitChunksV1(
+        uint256 _totalL1MessagesPoppedOverall,
+        bytes[] memory _chunks,
+        bytes calldata _skippedL1MessageBitmap
+    )
+        internal
+        view
+        returns (
+            bytes32 blobVersionedHash,
+            bytes32 _dataHash,
+            uint256 _totalL1MessagesPoppedInBatch
+        )
+    {
+        {
+            bytes32 secondBlob;
+            // Get blob's versioned hash
+            assembly {
+                blobVersionedHash := blobhash(0)
+                secondBlob := blobhash(1)
+            }
+            if (blobVersionedHash == bytes32(0)) revert ErrorNoBlobFound();
+            if (secondBlob != bytes32(0)) revert ErrorFoundMultipleBlob();
+        }
+
+        uint256 _chunksLength = _chunks.length;
+
+        // load `dataPtr` and reserve the memory region for chunk data hashes
+        uint256 dataPtr;
```

Comment:
It's a nitpick, but there are many different hashes so it's easy to get confused. So I suggest we call this `batchDataHashPtr` and the one inside `_commitChunk*` is `chunkDataHashPtr`.